src/Runner.Listener/JobDispatcher.cs

```diff
@@ -675,9 +685,128 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, string orc
             }
         }
 
-        public async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockToken, string orchestrationId, TaskCompletionSource<int> firstJobRequestRenewed, CancellationToken token)
+        public async Task RenewJobRequestAsync(Pipelines.AgentJobRequestMessage message, ServiceEndpoint systemConnection, int poolId, long requestId, Guid lockToken, string orchestrationId, TaskCompletionSource<int> firstJobRequestRenewed, CancellationToken token)
+        {
+            if (this._isRunServiceJob)
+            {
+                var runServer = await GetRunServerAsync(systemConnection);
+                await RenewJobRequestAsync(runServer, message.Plan.PlanId, message.JobId, firstJobRequestRenewed, token);
+            }
+            else
+            {
+                var runnerServer = HostContext.GetService<IRunnerServer>();
+                await RenewJobRequestAsync(runnerServer, poolId, requestId, lockToken, orchestrationId, firstJobRequestRenewed, token);
+            }
+        }
+
+        public async Task RenewJobRequestAsync(IRunServer runServer, Guid planId, Guid jobId, TaskCompletionSource<int> firstJobRequestRenewed, CancellationToken token)
```

Comment:
follow the same existing renewal without things like `RefreshConnectionAsync` etc., vssf realted